src/OrleansRuntime/Catalog/GrainCreator.cs

```diff
@@ -104,7 +106,7 @@ public Grain CreateGrainInstance(Type grainType, IGrainIdentity identity, Type s
             var logger = (factory as ILogConsistencyProvider)?.Log ?? storageProvider?.Log;
```

Comment:
This logic was always very weird for me, but it gets even more weird by making the contract official with a long Func<T1,T2,T3> that both parties need to know of. What if you just remove the need for passing the logger, and use the SP to assign one?